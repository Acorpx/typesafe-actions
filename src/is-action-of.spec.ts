import { createAction, isActionOf } from '.';

describe('isActionOf', () => {

  it('should succeed with same action', () => {
    const increment = createAction('INCREMENT');

    const isActionOfIncrement = isActionOf(increment);
    expect(isActionOfIncrement(increment())).toBeTruthy();
  });

  it('should fail with different action', () => {
    const increment = createAction('INCREMENT');
    const add = createAction('ADD');

    const isActionOfAdd = isActionOf(increment);
    expect(isActionOfAdd(add())).toBeFalsy();
  });

  it('should correctly assert type for EmptyAction', () => {
    const increment = createAction('INCREMENT');
    const isActionOfIncrement = isActionOf(increment);

    const action = { type: 'ERROR' };
    if (isActionOfIncrement(action)) {
      const a: { type: 'INCREMENT' } = action;
    }
  });

  it('should correctly assert type for FluxStandardAction', () => {
    const add = createAction('ADD',
      (amount: number) => ({ type: 'ADD', payload: amount }),
    );
    const isActionOfAdd = isActionOf(add);

    const action = { type: 'ERROR' };
    if (isActionOfAdd(action)) {
      const a: { type: 'ADD', payload: number } = action;
    }
  });

});
