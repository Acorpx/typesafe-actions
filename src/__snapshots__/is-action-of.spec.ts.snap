// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action 1`] = `"({ type: \\"WITH_MAPPED_PAYLOAD\\"; } & { payload: number; }) | ({ type: \\"WITH_MAPPED_PAYLOAD_META\\"; } & { payload: number; meta: string; })"`;

exports[`action 2`] = `"({ type: \\"WITH_MAPPED_PAYLOAD\\"; } & { payload: number; }) | ({ type: \\"WITH_MAPPED_PAYLOAD_META\\"; } & { payload: number; meta: string; })"`;

exports[`actual 1`] = `"Types.EmptyAction<\\"WITH_TYPE_ONLY\\">[]"`;

exports[`actual 2`] = `"(Types.EmptyAction<\\"WITH_TYPE_ONLY\\"> | Types.PayloadAction<\\"WITH_PAYLOAD\\", number>)[]"`;

exports[`actual 3`] = `"(Types.EmptyAction<\\"WITH_TYPE_ONLY\\"> | Types.PayloadAction<\\"WITH_PAYLOAD\\", number> | Types.PayloadMetaAction<\\"WITH_PAYLOAD_META\\", number, string>)[]"`;

exports[`actual 4`] = `"(Types.EmptyAction<\\"WITH_TYPE_ONLY\\"> | Types.PayloadAction<\\"WITH_PAYLOAD\\", number> | Types.PayloadMetaAction<\\"WITH_PAYLOAD_META\\", number, string> | ({ type: \\"WITH_MAPPED_PAYLOAD\\"; } & { payload: number; }))[]"`;

exports[`actual 5`] = `"(Types.EmptyAction<\\"WITH_TYPE_ONLY\\"> | Types.PayloadAction<\\"WITH_PAYLOAD\\", number> | Types.PayloadMetaAction<\\"WITH_PAYLOAD_META\\", number, string> | ({ type: \\"WITH_MAPPED_PAYLOAD\\"; } & { payload: number; }) | ({ type: \\"WITH_MAPPED_PAYLOAD_META\\"; } & { payload: number; meta: string; }))[]"`;
